caffe2/core/operator.h

```diff
@@ -453,6 +468,18 @@ class Operator : public OperatorBase {
     return OperatorBase::template Input<Tensor>(idx, type);
   }
 
+  inline Tensor* Output(
+      int idx,
+      const vector<int64_t>& dims,
+      const at::TensorOptions& options) {
+    if (options.device_opt() == at::nullopt) {
+      auto opt = options;
+      opt.device(context_.device());
```

Comment:
This usage will conflict with #12473. In this case, I'd appreciate it if you just rewrote this to be safe. It's pretty easy: instead of writing opt, write `TensorOptions(opt).device(context_.device())`.